### PR TITLE
[ARQ-1927] Make use of TCCL in ARQ ManagedContainer configurable

### PR DESCRIPTION
**Pull request by tdiesler**
https://issues.jboss.org/browse/ARQ-1927

### CODE REVIEW

File: protocols/jmx/src/main/java/org/jboss/arquillian/protocol/jmx/JMXProtocolConfiguration.java [hunk 1]
```diff
@@ -21,9 +21,18 @@
...
- * 
+ *
  * @author <a href="mailto:aslak@redhat.com">Aslak Knutsen</a>
- * @version $Revision: $
  */
 public class JMXProtocolConfiguration implements ProtocolConfiguration {
+
+    private boolean enableThreadContextClassLoader = true;
+
+    public boolean isEnableThreadContextClassLoader() {
```
Comment (aslakknutsen) [inline protocols/jmx/src/main/java/org/jboss/arquillian/protocol/jmx/JMXProtocolConfiguration.java]: Wouldn't this go into the JMX AS7 protocol impl? As it's fairly AS7 specific?

**Comment by aslakknutsen** [comment]
@tdiesler If we do something like this https://github.com/arquillian/arquillian-core/compare/master...aslakknutsen:82 we can make it backwards compatible as well.

Slightly hacky, but basically if 'this.getClass()' override runTestMethod(String, String) then call that instead of the Map version. Else call just call the internal run method.

Specifically this part: https://github.com/arquillian/arquillian-core/compare/master...aslakknutsen:82#diff-3470c459de35b2bb8b96bdff42341276R129

**Comment by tdiesler** [comment]
Yes, although I believe it is easier to go through the extended JMX protocols and add the new parameter. Even if there are extenders that we don't control, it would be easy for them to make that change if they should choose to update arq-core. The resulting error would indicate what there is to to.

Anyway, thanks for making this effort - your call if you want the compatibility.

**Comment by aslakknutsen** [comment]
@tdiesler Of course. This is not about how easy it is to support the new version, but to be able to use the new version on old containers that have no updated. e.g. still be able to test EAP with the new Core v.

**Comment by tdiesler** [comment]
Is this waiting on something? i.e. will it be part of 1.1.8.Final?

**Comment by aslakknutsen** [comment]
no, just me. yes.

**Comment by aslakknutsen** [comment]
Pushed upstream https://github.com/arquillian/arquillian-core/commit/80d481bb0c0999b0c815a4c55b0e420682db4b6d & https://github.com/arquillian/arquillian-core/commit/d1f223a10bf8b0f89ae2875abd2b6719b8109b17

**Pull request merged**